src/Compilers/Core/Portable/SourceGeneration/GeneratorState.cs

```diff
@@ -2,31 +2,103 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
 
 #nullable enable
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// Represents the current state of a generator
+    /// </summary>
+    /// <remarks>
+    /// A generator state is essentially a small state machine:
```

Comment:
Should we have an enum tracking what state we're actually in? It would allow us to assert that we're only making valid state transitions.

Reply:
I'm not sure that's needed, especially as we really only have a couple of simple states. Maybe if it got more complicated, but it seems like a lot of overhead for something that's easily verifyable and only used by a single class.

---
In reply to: [466072377](https://github.com/dotnet/roslyn/pull/46296#discussion_r466072377) [](ancestors = 466072377)